src/ui/views/ManualDownload.svelte

```diff
@@ -1,23 +1,102 @@
+
+  // Pick directory and construct full file path
+  async function selectDirectoryAndConstructPath() {
+    errorMessage = ""; // Clear previous errors
+    
+    try {
+      const result = await ipcRenderer.invoke('show-file-dialog', {
+        properties: ['openDirectory'],
+        defaultPath: require('os').homedir() + '/Downloads',
+        title: `Select Directory containing ${$manualDownloadFileData.name}`
+      });
+      
+      if (!result.canceled && result.filePaths && result.filePaths.length > 0) {
+        selectedDirectory = result.filePaths[0];
+        
+        // Sanitize and validate filename
+        const expectedFilename = path.basename($manualDownloadFileData.name);
+        
+        if (!expectedFilename || expectedFilename === '.' || expectedFilename === '..') {
+          errorMessage = "Invalid filename provided";
+          return;
+        }
+        
+        // Join directory path + expected filename
+        constructedFilePath = path.join(selectedDirectory, expectedFilename);
+        
+        console.log("Selected directory:", selectedDirectory);
+        console.log("Expected filename:", expectedFilename);
+        console.log("Constructed path:", constructedFilePath);
+        
+        // Verify file exists and is readable
+        if (fs.existsSync(constructedFilePath)) {
+          try {
+            fs.accessSync(constructedFilePath, fs.constants.R_OK);
+            console.log("File found and readable at:", constructedFilePath);
+          } catch (accessError) {
+            errorMessage = "File exists but cannot be read. Check permissions.";
+            console.error("File not readable:", accessError);
+          }
+        } else {
+          errorMessage = `File "${expectedFilename}" not found in the selected directory. Please ensure the file has been downloaded to this location.`;
+          console.log("File not found at:", constructedFilePath);
+        }
+      }
+    } catch (error) {
+      errorMessage = `Error selecting directory: ${error.message}`;
+      console.error("Error:", error);
+    }
+  }
 
   function handleManualDownloadButton() {
+    if (!constructedFilePath) {
+      alert("Please select the directory first.");
+      return;
+    }
+    
+    // Verify file exists before sending
+    if (!fs.existsSync(constructedFilePath)) {
+      alert(`File not found: ${constructedFilePath}\n\nPlease ensure the file has been downloaded to the selected directory.`);
+      return;
+    }
+    
+    // Verify file is readable
+    try {
+      fs.accessSync(constructedFilePath, fs.constants.R_OK);
+    } catch (error) {
+      alert(`Cannot read file: ${constructedFilePath}\n\nPlease check file permissions.`);
+      return;
+    }
+    
     checkingFile = true;
+    
     ipcRenderer.once("user:manual_download:check", (event, ok) => {
       checkingFile = false;
+      if (!ok) {
+        errorMessage = "File verification failed. Please try again.";
+      }
```

Comment:
This is already handled in the function that sent this event, it will throw and exit the whole flow (meaning this message won't even show up)